src/Facility.Definition/ServiceInfo.cs

```diff
@@ -155,6 +164,68 @@ ServiceMemberInfo DoExcludeTag(ServiceMemberInfo member)
 			}
 		}
 
+		private static void EnsureProperValidateUsage(ServiceTypeInfo type, ServiceFieldInfo field)
+		{
+			if (field.Validation == null) return;
+			var validateAttribute = field.TryGetAttribute("validate")!;
+
+			switch (type.Kind)
+			{
+				case ServiceTypeKind.Enum:
+				{
+					foreach (var validateAttributeParameter in validateAttribute.Parameters)
+						field.AddValidationError(ServiceDefinitionUtility.CreateUnexpectedAttributeParameterError(validateAttribute.Name, validateAttributeParameter));
```

Comment:
It might be better to report the error on the attribute or parameter rather than the field.